ratis-server/src/main/java/org/apache/ratis/server/storage/SnapshotManager.java

```diff
@@ -173,9 +172,10 @@ public void installSnapshot(InstallSnapshotRequestProto request, StateMachine st
     }
   }
 
-  private MessageDigest newMd5Digest() throws IOException {
+  private void renewMd5Digest() throws IOException {
     try {
-      return MessageDigest.getInstance("MD5");
+      digester =  MessageDigest.getInstance("MD5");
+      digester.reset();
```

Comment:
I think calling `MessageDigest#reset` here is unnecessary since a newly instantiated `MessageDigest` should already be "reset".

Reply:
Thank you for your reminder